### PR TITLE
Removes `export env` command

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

while reading https://hackmd.io/@nucore/r1zilyw6q to see what's the future about module environment, I sopt that most of thing are done.

And export env doesn't work with `source-env` command.

So this pr is trying to help to remove `export env` command, and adjust relative tests and examples

About three more ignored tests:
1. use_env_import_after_hide: it's more about `hide-env`, `overlay`, `export-env` commands
2. hides_all_envs_within_scope: it's more about `hide-env`, `overlay`, `export-env` commands
3. remove_overlay_keep_discard_overwritten_env: it's more about `overlay` with `export-env` issue itself, so I think it's ok to remove it

They are existing issue about the communication of `overlay`, `export-env` and `hide-env` commands, I think we can keep it ignored for now, and fix them in separate pr.
But I think it's also ok to wait these relative issue to be fixed firstly.

# Tests

Make sure you've done the following:

- [X] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [X] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [X] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

Make sure you've run and fixed any issues with these commands:

- [X] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [X] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [X] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

**Comment by kubouch** [comment]
Thanks! I purposefully kept it in for the next release for compatibility reasons. We can land it after the next release.

File: src/tests/test_hiding.rs [hunk 1]
```diff
@@ -417,6 +383,7 @@ fn use_def_import_after_hide() -> TestResult {
 }
 
 #[test]
```
Comment (kubouch) [inline src/tests/test_hiding.rs:385]: We can delete this test

**Comment by kubouch** [comment]
I'm not sure about the ignored tests. Could you try converting them to use the `export-env` and `overlay use` like the other tests? I'd like to see what errors they bring.

**Comment by kubouch** [comment]
Oh, and the whole "Cleanup" part from the hackmd I'm planning to do after the release so people have time to transition and fix their scripts until the next-next release.

**Comment by WindSoilder** [comment]
> I'm not sure about the ignored tests. Could you try converting them to use the export-env and overlay use like the other tests? I'd like to see what errors they bring.

Actually I've converted them to use `export-env` and `overlay use`, but after manually testing, it doesn't work as test expected:
## hides_all_envs_within_scope 
```
> module spam { export-env { let-env foo = "bar" } }; let-env foo = "foo"; overlay use spam; hide-env foo; $env.foo
foo
```
It doesn't failed with `did you mean` message

## use_env_import_after_hide
```
> module spam { export-env {let-env foo = "foo" } }; overlay use spam; hide-env foo; overlay use spam; $env.foo
Error: nu::shell::name_not_found (link)

  × Name not found
   ╭─[entry #1:1:1]
 1 │ module spam { export-env {let-env foo = "foo" } }; overlay use spam; hide-env foo; overlay use spam; $env.foo
   ·                                                                                                           ─┬─
   ·                                                                                                            ╰── did you mean 'HOME'?
   ╰────

```
It doesn't output `foo`.

## remove_overlay_keep_discard_overwritten_env
```
❯ cat spam.nu
export def foo [] { "foo" }

export alias bar = "bar"

export-env { let-env BAZ = "baz" } }

~
❯ overlay use spam.nu; let-env BAZ = `bagr`; overlay hide --keep-custom spam; $env.BAZ
bagr
```
It doesn't failed with `did you mean` message

**Comment by kubouch** [comment]
@WindSoilder If you don't mind, I'll push some changes to your PR. I'd like to check the tests and also investigate the `--keep-custom` flag and it's easier to for me to fix up these things myself.

**Comment by kubouch** [comment]
I went through the tests again because the `use` command adopted the `source-env` functionality. I also finished all the environment cleanup.

The virtualenv tests are failing because `hide` no longer hides environment variables. We need to wait until https://github.com/pypa/virtualenv/pull/2422 is merged.

**Pull request merged**